cukinia

```diff
@@ -877,6 +883,66 @@ _cukinia_knoerror()
 	return $ret
 }
 
+# _cukinia_network_interface_up: check if a network interface is up
+# arg1: interface name
+_cukinia_network_interface_up() {
+	local interface="$1"
+
+	_cukinia_prepare "Checking if interface \"$interface\" is ${__not:+NOT }up"
+
+	ip link show "$interface" up >/dev/null 2>&1 && return 0 || return 1
+}
+
+# _cukinia_network_interface_ipv4: check if an interface has an IPv4 address
+# arg1: interface name
+_cukinia_network_interface_ipv4() {
+	local interface="$1"
+
+	_cukinia_prepare "Checking if interface \"$interface\" has ${__not:+NO }IPv4 address"
+
+	ip -4 -f inet addr show dev "$interface" | grep -q inet && return 0 || return 1
+}
+
+# _cukinia_network_interface_global_ipv6: check if a network interface has a global IPv6 address
+# arg1: interface name
+_cukinia_network_interface_global_ipv6() {
+	local interface="$1"
+
+	_cukinia_prepare "Checking if interface \"$interface\" has ${__not:+NO }global IPv6 address"
+
+	ip -o -6 address show dev "$interface" scope global | grep -q "inet6" && return 0 || return 1
+}
+
+# _cukinia_network_interface_dhcp_ipv4: check if an interface has an IPv4 address assigned via DHCP
+# arg1: interface name
+_cukinia_network_interface_dhcp_ipv4() {
+	local interface="$1"
+
+	_cukinia_prepare "Checking if interface \"$interface\" has ${__not:+NO }IPv4 address assigned via DHCP"
+
+	ip -4 -f inet addr show dev "$interface" | grep -q 'scope global dynamic' && return 0 || return 1
```

Comment:
Which guarantee do we have that no extra flags can be inserted between "scope" "global" and "dynamic"?

Reply:
As per current code base there is no chance of extra flags but in future there may be chance. so changed the syntax.